pandas/_testing/__init__.py

```diff
@@ -207,8 +207,10 @@ def box_expected(expected, box_cls, transpose=True):
         if transpose:
             # for vector operations, we need a DataFrame to be a single-row,
             #  not a single-column, in order to operate against non-DataFrame
-            #  vectors of the same length.
+            #  vectors of the same length. But convert to two rows to avoid
+            #  single-row special cases in datetime arithmetic
             expected = expected.T
+            expected = pd.concat([expected] * 2, ignore_index=True)
```

Comment:
@jbrockmendel by making this a (2, x) DataFrame instead of a (1, x) shaped DataFrame, it avoids some special cases in the tests where we now often have "if box is DataFrame -> don't raise a PerformanceWarning" (which is quite confusing, because it has nothing to do with it being a DataFrame, but rather with the column arrays being length 1)

Reply:
seems Mostly Harmless i guess